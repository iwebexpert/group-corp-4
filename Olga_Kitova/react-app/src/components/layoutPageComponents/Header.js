import React, { useContext, useEffect} from 'react'
import {useDispatch, useSelector} from 'react-redux'
import Toolbar from '@mui/material/Toolbar'
import IconButton from '@mui/material/IconButton'
import Link from '@mui/material/Link'
import {Link as RouterLink} from 'react-router-dom'
import { useTheme } from '@mui/material/styles'
//Context
import { Context } from 'context/Context'
import LightModeIcon from '@mui/icons-material/LightMode'
import NightlightIcon from '@mui/icons-material/Nightlight'
import { Tooltip } from '@mui/material'
import {authService} from 'services/auth/authService'
import { getPagesFetch } from 'actions/actionsPages'


export default function Header() {
  const theme = useTheme();
  const {toggleTheme} = useContext(Context)
  const user = authService?.currentUserValue
  const name = user?.name || ''
  const role = user?.role || null
  const dispatch = useDispatch()
  const headers = useSelector(state => state.pages.headerPages)

  useEffect(() => {
     dispatch(getPagesFetch())
  },[])

  return (
    <React.Fragment>
      <Toolbar sx={{ borderBottom: 1, borderColor: 'divider', justifyContent: 'flex-end' }}>
        <div>{name ? `Hello, ${name}` : null}</div>
        <br />
        <Tooltip
            title={
              theme.palette.mode !== 'light' ? 'Choose a light theme' : 'Choose a dark theme'
            }
          >
            <IconButton color="inherit" onClick={() => {
              toggleTheme.toogleThemeMode()
            }}>
            {theme.palette.mode !== 'light' ? <LightModeIcon /> : <NightlightIcon />}
            </IconButton>
            </Tooltip>
{role ?  <Link href='/logout' sx={{marginLeft: '1rem'}} variant="outlined" size="small">
          Log Out</Link> : <Link href='/login' sx={{marginLeft: '1rem'}} variant="outlined" size="small">
          Log In</Link>}
      </Toolbar>
      <Toolbar
        component="nav"
        variant="dense"
        sx={{ justifyContent: 'center', overflowX: 'auto', paddingBottom: '2rem', paddingTop: '1rem'}}
      >
  {role && (<>
        <Link 
        color="inherit"
        noWrap
        variant="body2"
        sx={{ p: 1, flexShrink: 0 }}
        href="/admin/profile">Profile</Link>
        <Link 
        color="inherit"
        noWrap
        variant="body2"
        sx={{ p: 1, flexShrink: 0 }}
        href="/admin">Dashboard</Link></>)
        }
        {headers.map((section) => (
          <Link
            component={RouterLink}
            color="inherit"
            noWrap
            key={section.title}
            variant="body2"
            to={`/page/${section.url}`}
            sx={{ p: 1, flexShrink: 0 }}
          >
            {section.title}
          </Link>
        ))}
      </Toolbar>
    </React.Fragment>
  );
}