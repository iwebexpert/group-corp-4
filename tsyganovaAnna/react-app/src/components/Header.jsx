import React from 'react'

import { AppBar, Typography, Toolbar, Tooltip, IconButton } from '@mui/material'
import LightModeIcon from '@mui/icons-material/LightMode'
import NightlightIcon from '@mui/icons-material/Nightlight'
import MenuIcon from '@mui/icons-material/Menu'
import { useTheme } from '@mui/material/styles'

import { ThemeModeContext } from '../contexts/ThemeContext'
import { authService } from '../services/auth/authService'
import AccountMenu from './AccountMenu'

export default function Header({ HandleClick }) {
  const theme = useTheme()
  const { colorMode } = React.useContext(ThemeModeContext)
  const userExist = authService.currentUserExist
  return (
    <AppBar position="static">
      <Toolbar>
        <Tooltip title="Menu">
          <IconButton
            color="inherit"
            aria-label="open drawer"
            onClick={HandleClick}
            edge="start"
            sx={{
              marginRight: 5,
              ...(!userExist && { display: 'none' }),
            }}
          >
            <MenuIcon />
          </IconButton>
        </Tooltip>
        <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
          Learn project
        </Typography>
        <Tooltip title={`${theme.palette.mode} mode`}>
          <IconButton color="inherit" onClick={colorMode.toogleColorMode}>
            {theme.palette.mode !== 'light' ? <LightModeIcon /> : <NightlightIcon />}
          </IconButton>
        </Tooltip>
        {userExist && <AccountMenu user={authService.currentUser} logout={authService.logout} />}
      </Toolbar>
    </AppBar>
  )
}
