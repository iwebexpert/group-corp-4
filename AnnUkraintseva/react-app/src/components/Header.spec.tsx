import React from "react"
import Header from "./LoadedForm"

import * as renderer from 'react-test-renderer'
import {test, expect} from '@jest/globals'

import {shallow, mount} from 'enzyme'

test('Copyright should be renderer', ()=>{
  const testRenderer = renderer.create(<Header/>)
  const treeJson = testRenderer.toJSON()
  expect(treeJson). toMatchSnapshot()

})