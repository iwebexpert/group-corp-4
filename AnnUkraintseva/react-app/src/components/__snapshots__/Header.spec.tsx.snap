// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright should be renderer 1`] = `
<div
  className="preloader"
>
  <div
    className="preloader__row"
  >
    <div
      className="preloader__item"
    />
    <div
      className="preloader__item"
    />
  </div>
</div>
`;
